Awful.apk/src/main/java/com/ferg/awfulapp/NavigationEvent.kt

```diff
@@ -28,7 +29,7 @@ sealed class NavigationEvent(private val extraTypeId: String) {
      * Defaults to opening the main activity, override this for events handled by other activities.
      */
     protected open fun activityIntent(context: Context): Intent =
-            context.intentFor(ForumsIndexActivity::class.java).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+            context.intentFor("com.ferg.awfulapp.ForumsIndexActivity."+AwfulPreferences.getInstance().launcherIcon).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
```

Comment:
I'd revert all the changes in NavigationEvent - it works without them, and it feels like involving the launcher icon and aliases is adding complexity that could cause weird issues later. Unless there's a reason to keep these changes I'd avoid it, honestly